lib/src/virtual_directory.dart

```diff
@@ -181,110 +178,112 @@ class VirtualDirectory {
+          if (start <= end) {
+            if (end >= length) {
               end = length - 1;
-            } else {
-              start = int.parse(matches[1]);
-              end = matches[2].isEmpty ? length - 1 : int.parse(matches[2]);
             }
-            // If the range is syntactically invalid the Range header
-            // MUST be ignored (RFC 2616 section 14.35.1).
-            if (start <= end) {
-              if (end >= length) {
-                end = length - 1;
-              }
 
-              if (start >= length) {
-                response
-                  ..statusCode = HttpStatus.requestedRangeNotSatisfiable
-                  ..close();
-                return;
-              }
+            if (start >= length) {
+              response.statusCode = HttpStatus.requestedRangeNotSatisfiable;
+              await response.close();
+              return;
+            }
+
+            // Override Content-Length with the actual bytes sent.
+            response.headers
+                .set(HttpHeaders.contentLengthHeader, end - start + 1);
+
+            // Set 'Partial Content' status code.
+            response
+              ..statusCode = HttpStatus.partialContent
+              ..headers.set(
+                  HttpHeaders.contentRangeHeader, 'bytes $start-$end/$length');
 
-              // Override Content-Length with the actual bytes sent.
-              response.headers
-                  .set(HttpHeaders.contentLengthHeader, end - start + 1);
-
-              // Set 'Partial Content' status code.
-              response
-                ..statusCode = HttpStatus.partialContent
-                ..headers.set(HttpHeaders.contentRangeHeader,
-                    'bytes $start-$end/$length');
-
-              // Pipe the 'range' of the file.
-              if (request.method == 'HEAD') {
-                response.close();
-              } else {
-                file
+            // Pipe the 'range' of the file.
+            if (request.method == 'HEAD') {
+              await response.close();
+            } else {
+              try {
+                await file
                     .openRead(start, end + 1)
                     .cast<List<int>>()
-                    .pipe(_VirtualDirectoryFileStream(response, file.path))
-                    .catchError((_) {
-                  // TODO(kevmoo): log errors
-                });
+                    .pipe(_VirtualDirectoryFileStream(response, file.path));
+              } catch (_) {
+                // TODO(kevmoo): log errors
               }
-              return;
             }
+            return;
           }
         }
+      }
 
-        response.headers.set(HttpHeaders.contentLengthHeader, length);
-        if (request.method == 'HEAD') {
-          response.close();
-        } else {
-          file
+      response.headers.set(HttpHeaders.contentLengthHeader, length);
+      if (request.method == 'HEAD') {
+        await response.close();
```

Comment:
Technically I don't think we have to await these - they were effectively unawaited before.

Reply:
In the one place where it seemed to me like it mattered I added a little `_unawaited` utility. For a bunch of these I kept the `await` since it didn't seem like it mattered one way or the other - the method returns `void` so waiting for more work won't matter, nothing will be waiting for the overall result. I don't _think_ I left in the ones that would impact the interleaving of work.